### DISCUSSION TITLE
Update Vehicle status with MyBMW (new API)

### DISCUSSION THREAD

**Discussion by FilipDem** [bimmerconnected/bimmer_connected#343] [Feature request] [answered]
Hi @rikroe! First of all a big thanks!
Just an suggestion... I am using this library to integrate with Domoticz (not HA).

I want to get regularly updates, but perhaps it is not useful to start the complete authentication phase as long as the token is valid. And I see that this can only be done by using the function _get_vehicles(). Seen the underscore it is normally an internal function however I would suggest to make a public one so that it can be called directly.

Moreover the function _get_vehicles() cannot be used as it appends always all vehicles to the list... So the list gets bigger and bigger with same vehicles...

Therefor another public function to get just the new update vehicle information would be nice.

Thanks,
Filip

**Comment by rikroe** [comment]
Hi Filip,

I was thinking about your issue again and have a question: How are you integrating it into Domoticz? Are you just calling our CLI?

If so, I could imaging adding a global parameter to the CLI that stores token & token expiry to a file (that you could put wherever needed) and that of course could be loaded each time before making any other calls.

**Comment by FilipDem** [comment, marked as answer]
Hi, I am not using the CLI, but the python APIs...
You can find the implemenation on [plugin.py](https://github.com/FilipDem/Domoticz-BMW-plugin/blob/main/plugin.py). Mainly the function `handleTasks` and `updateVehicleStatus` are using the APIs... So recreation of the tokens is managed by the bimmer_connected as far as I understood.

The complete documentation/instructions for Domoticz can be found at [Domoticz Plugin](https://github.com/FilipDem/Domoticz-BMW-plugin).

But thanks again for the work and the suggestion to help.

Filip

**Comment by rikroe** [comment]
If I understand your code correctly, you are storing `ConnectedDriveAccount` in `self.myBMW` ([plugin.py#L224](https://github.com/FilipDem/Domoticz-BMW-plugin/blob/main/plugin.py#L224)).

As all other functions are executed from `self.myBMW`, the token lifetime will be checked and only a full login will be executed if the token has expired. That is part of `_get_oauth_token()` ([account.py#L88-L90](https://github.com/bimmerconnected/bimmer_connected/blob/master/bimmer_connected/account.py#L88-L90)).

So basically, everything should be all right?

There was the issue with `_get_vehicles()` not updating vehicles though. But I think that was fixed in one of the beta releases before 0.8.0 was published to PyPi.

**Comment by FilipDem** [comment]
Indeed this is all solved and is working very well. 
Thanks.
